src/plugins/resolver/filename.c

```diff
@@ -406,11 +422,26 @@ static int elektraResolveSpec (resolverHandle * p, Key * warningsKey ELEKTRA_UNU
 	}
 	else
 	{
-		strcpy (p->filename, KDB_DB_SPEC);
-		strcat (p->filename, "/");
-		strcat (p->filename, p->path);
+		if (KDB_DB_SPEC[0] == '~')
```

Comment:
not really, this version would only work after dropping CMAKE_INSTALL_PREFIX for pathnames starting with `~`
i don't really see an other way of doing it inside the resolver